deepspeed/profiling/flops_profiler/profiler.py

```diff
@@ -157,10 +158,26 @@ def reset_profile(self):
         Adds or resets the extra attributes.
         """
 
+        def moe_expert_parallelism(param):
+            """
+            Return whether the layer is an MoE layer, and its expert parallelism size if so
+            """
+            prefix = 'ep_size_'
+            if getattr(param, "group_name", "").startswith(prefix):
+                return True, int(param.group_name[len(prefix):])
+            return False, 0
+
         def add_or_reset_attrs(module):
+            parameters = []
+            for p in module.parameters():
+                is_moe, expert_parallelism = moe_expert_parallelism(p)
+                parameters.append((p.numel(), is_moe, expert_parallelism))
             module.__flops__ = 0
             module.__macs__ = 0
-            module.__params__ = sum(p.numel() for p in module.parameters())
+            module.__params__ = sum(count for count, is_expert, _ in parameters if not is_expert)
+            module.__expert_params__ = sum(count for count, is_expert, _ in parameters if is_expert)
```

Comment:
Do we still need this?

Reply:
Yes, see the final calculation

Reply:
I see, we still use it for per-module breakdown, right?